File: examples/roots_real_tower.py
#
# jython examples for jas.
# $Id$
#

import sys;

from java.lang import System
from java.lang import Integer

from jas import Ring, PolyRing
from jas import QQ, DD, EF
from jas import terminate
from jas import startLog


# polynomial examples: real root tower over Q

r = EF(QQ()).realExtend("q","q^3 - 3", "[1,2]").realExtend("w", "w^2 - q", "[1,2]").realExtend("s", "s^5 - 2", "[1,2]").polynomial("x").build();

print "Ring: " + str(r);
print;

[one,q,w,s,x] = r.gens();
print "one   = " + str(one);
print "q     = " + str(q);
print "w     = " + str(w);
print "s     = " + str(s);
print "x     = " + str(x);
print;

f = x**2 - w * s;


print "f = ", f;
print;

startLog();

t = System.currentTimeMillis();
R = r.realRoots(f);
t = System.currentTimeMillis() - t;
#print "R = ", R;
print "R = ", [ a.elem.ring.getRoot() for a in R ];
print "real roots time =", t, "milliseconds";

#sys.exit();

eps = QQ(1,10) ** (DD().elem.DEFAULT_PRECISION);
print "eps = ", eps;

t = System.currentTimeMillis();
R = r.realRoots(f,eps);
t = System.currentTimeMillis() - t;
#print "R = ", [ str(r) for r in R ];
print "R = ", [ a.elem.decimalMagnitude() for a in R ];
print "real roots time =", t, "milliseconds";
print;

#startLog();
terminate();

File: examples/roots_real_tower.rb
#
# jruby examples for jas.
# $Id$
#

require "examples/jas"

# polynomial examples: real roots tower over Q

r = EF.new(QQ()).realExtend("q","q^3 - 3", "[1,2]").realExtend("w", "w^2 - q", "[1,2]").realExtend("s", "s^5 - 2", "[1,2]").polynomial("x").build();

puts "Ring: " + str(r);
puts;

one,q,w,s,x = r.gens();


f = x**2 - w * s;

puts "f = " + f.to_s;
puts;


startLog();

t = System.currentTimeMillis();
rr = r.realRoots(f);
t = System.currentTimeMillis() - t;
#puts "rr = " + str(rr);
puts "rr = " + str(rr.map{ |a| str(a.elem.ring.getRoot())+"," }); 
puts "real roots time = " + str(t) + " milliseconds";

eps = QQ(1,10) ** (BigDecimal::DEFAULT_PRECISION);
puts "eps = " + str(eps);

t = System.currentTimeMillis();
rr = r.realRoots(f,eps);
t = System.currentTimeMillis() - t;
#puts "rr = ", [ str(r) for r in rr ];
puts "rr = " + str(rr.map{ |a| str(a.elem.decimalMagnitude())+"," }); 
puts "real roots time = " + str(t) + " milliseconds";

#startLog();
terminate();
